src/tests/java/htsjdk/samtools/sra/SRATest.java

```diff
@@ -0,0 +1,551 @@
+
+        SamReader reader = SamReaderFactory.make().validationStringency(ValidationStringency.SILENT).open(
+                SamInputResource.of(new SRAAccession(acc))
+        );
+
+        final SAMRecordIterator samRecordIterator = ((SamReader.Indexing) reader).iterator(new BAMFileSpan(chunks));
+
+        int countAlignments = 0, countUnalignments = 0;
+        while (true) {
+            boolean hasRecord = samRecordIterator.hasNext();
+            SAMRecord record = null;
+            try {
+                record = samRecordIterator.next();
+                Assert.assertTrue(hasRecord); // exception is not thrown if we came to this point
+            } catch (NoSuchElementException e) {
+                Assert.assertFalse(hasRecord);
+            }
+
+            Assert.assertEquals(hasRecord, record != null);
+
+            if (record == null) {
+                break;
+            }
+
+            if (record.getReadUnmappedFlag()) {
+                countUnalignments++;
+            } else {
+                countAlignments++;
+            }
+        }
+
+        Assert.assertEquals(numberAlignments, countAlignments);
+        Assert.assertEquals(numberUnalignments, countUnalignments);
+    }
+
+    @DataProvider(name = "testGroups")
+    public Object[][] createDataForGroups() {
+        return new Object[][] {
+            {"SRR822962", Arrays.asList("GS54389-FS3-L08", "GS54389-FS3-L08", "GS57511-FS3-L08", "GS54387-FS3-L02",
+                                        "GS54387-FS3-L01", "GS57510-FS3-L01", "GS57510-FS3-L03", "GS54389-FS3-L07",
+                                        "GS54389-FS3-L05", "GS54389-FS3-L06", "GS57510-FS3-L02", "GS57510-FS3-L04",
+                                        "GS54387-FS3-L03", "GS46253-FS3-L03")},
+            {"SRR2096940", Arrays.asList("_")}
+        };
+    }
+
+    @Test(dataProvider = "testGroups")
+    public void testGroups(String acc, List<String> groups) {
+        if (!SRAAccession.isSupported()) return;
+
+        SamReader reader = SamReaderFactory.make().validationStringency(ValidationStringency.SILENT).open(
+                SamInputResource.of(new SRAAccession(acc))
+        );
+
+        final SAMRecordIterator samRecordIterator = reader.iterator();
+
+        SAMFileHeader header = reader.getFileHeader();
+        Set<String> headerGroups = new TreeSet<String>();
+        for (SAMReadGroupRecord group : header.getReadGroups()) {
+            Assert.assertEquals(group.getReadGroupId(), group.getId());
+            headerGroups.add(group.getReadGroupId());
+        }
+
+        Assert.assertEquals(headerGroups.containsAll(groups), true, "Header groups are different from expected");
+
+        Set<String> requiredGroups = new HashSet<String>(groups);
+        requiredGroups.removeAll(headerGroups);
+
+        Assert.assertEquals(requiredGroups.size(), 0, "Following groups are missing in SRA archive header: " +
+                requiredGroups.toString());
```

Comment:
Would be simpler to just convert `groups` to a `Set` and rely on `Set.equals()` here.